modules/docs/docgen/docParser.ts

```diff
@@ -0,0 +1,1777 @@
+   * defaultValue of things. If the initializer of a property is an identifier and that identifier
+   * is an exported symbol, we'll assign it a symbol.
+   *
+   * For example:
+   * ```ts
+   * const a = b
+   * ```
+   *
+   * In this example, `b` is the identifier because it is a value reference to something else.
+   */
+  if (t.isIdentifier(node)) {
+    if (isExportedSymbol(checker, node)) {
+      const symbol = getSymbolFromNode(checker, node);
+      const declaration = getValueDeclaration(symbol);
+      return {
+        kind: 'symbol',
+        name: node.text,
+        fileName: declaration?.getSourceFile().fileName,
+        value: node.text,
+      };
+    }
+    if (node.text === 'undefined') {
+      return {kind: 'primitive', value: 'undefined'};
+    }
+  }
+
+  /**
+   *
+   */
+  if (t.isParameter(node)) {
+    const type = checker.getTypeAtLocation(node);
+    const symbol = getSymbolFromNode(checker, node);
+    const jsDoc = findDocComment(checker, symbol);
+    const isRequired = node.questionToken
+      ? false
+      : node.initializer
+      ? false
+      : symbol
+      ? !isOptional(symbol) && !includesUndefined(type)
+      : false;
+
+    const typeInfo = node.type
+      ? getValueFromNode(parser, node.type)
+      : getValueFromType(parser, type) || unknownValue(safeGetText(checker, node));
+
+    const defaultValue = node.initializer ? getValueFromNode(parser, node.initializer) : undefined;
+
+    /**
+     * Set default values if an object binding pattern is found. We do this at the Parameter level,
+     * because we have all the info here.
+     *
+     * For example:
+     * ```ts
+     * function A({ a = 'a', b}: Params)
+     * ```
+     *
+     * In this example, the ObjectBindingPattern is `{ a = 'a', b }`
+     */
+    if (t.isObjectBindingPattern(node.name)) {
+      const defaults = getDefaultsFromObjectBindingParameter(parser, node);
+      if (typeInfo.kind === 'object') {
+        typeInfo.properties.forEach(p => {
+          if (!p.defaultValue && defaults[p.name]) {
+            p.defaultValue = defaults[p.name];
+          }
+        });
+      }
+    }
+
+    return {
+      kind: 'parameter',
+      name: symbol?.name || '',
+      defaultValue,
+      type: typeInfo,
+      required: isRequired,
+      rest: !!node.dotDotDotToken,
+      ...jsDoc,
+    };
+  }
+
+  /**
+   * A call expression is an expression calling a function. In this case, we want to get the signature
+   * and get a type for the return type
+   */
+  if (t.isCallExpression(node)) {
+    const type = checker.getTypeAtLocation(node);
+    const value = getValueFromType(parser, type);
+    if (value) {
+      return value;
+    }
+  }
+
+  const symbol = getSymbolFromNode(checker, node);
+  if (!symbol) {
+    return unknownValue(safeGetText(checker, node));
+  }
+  const declaration = getValueDeclaration(symbol);
+  (declaration && t.getKindNameFromNode(declaration)) || symbol.name;
```

Comment:
What's going on with this expression? It doesn't seem to serve a purpose, or it was modified unexpectedly.

Reply:
It originally had a `//?` at the end for WallabyJS logging. I'll remove it.